src/power_grid_model/core/power_grid_model.py

```diff
@@ -385,13 +403,18 @@ def calculate_power_flow(
         error_tolerance: float = 1e-8,
         max_iterations: int = 20,
         calculation_method: Union[CalculationMethod, str] = CalculationMethod.newton_raphson,
-        update_data: Optional[Dict[str, Union[np.ndarray, Dict[str, np.ndarray]]]] = None,
+        update_data: Optional[
+            Union[
+                Dict[str, Union[np.ndarray, Dict[str, np.ndarray]]],
+                Dataset,
+            ]
+        ] = None,
```

Comment:
@mgovers I implemented it this way as this is a user available function: Basically it can take `Dataset` as defined with `ComponentType` but also as defined with `Literal`. I had to do it this way for mypy to like it when users decide to use `Literal`s (backwards support) instead of the new datatypes.